app/controllers/sitcoms_controller.rb

```diff
@@ -1,8 +1,11 @@
 class SitcomsController < ApplicationController
-  before_action :authenticate_user!, only: [:new, :create]
+  before_action :authenticate_user!, only:
+  [:new, :create, :edit, :update, :destroy]
+  before_action :fetch_sitcom, only: [:show, :edit, :destroy]
 
   def index
-    @sitcoms = Sitcom.all.limit(20)
+    sitcoms = Sitcom.all.limit(20)
+    @sitcoms = sitcoms.sort_by { |sitcom| sitcom.start_year }.reverse
```

Comment:
This should be refactored.  Try something like this:
```ruby
@sitcoms = Sitcom.order(start_year: :desc)
```
http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-order